### ISSUE TITLE
Builtins should not be treated as undeclared by 'no-undef' rule

### ISSUE THREAD

**Issue by mamacdon** [eslint/eslint#185]
Running the `no-undef` rule on this code:

``` js
if (!Object.isPrototypeOf.call(Array.prototype, [])) {
    throw new Error("yikes");
}
```

produces warnings:

```
Warning - 'Object' is not defined.
Warning - 'Array' is not defined.
Warning - 'Error' is not defined.
```

I expected this to pass with no warnings, since all the above objects are required by the spec.

(Note: I think this is separate from issue #180, since the stuff I'm referring to here is specified as members of the global object itself, not the global object's [[Prototype]], nor `Object.prototype`.)
